library/ssl_msg.c

```diff
@@ -851,7 +851,7 @@ static void ssl_extract_add_data_from_record(unsigned char *add_data,
     *add_data_len = (size_t) (cur - add_data);
 }
 
-#if defined(MBEDTLS_SSL_HAVE_AEAD)
```

Comment:
Sorry to pop in late but I don't think we should abandon the macro `MBEDTLS_SSL_HAVE_AEAD`. I think we should keep it, only now defined solely in terms of PSA availability rather than PSA or legacy.

We might rename it to follow existing names: we already have `MBEDTLS_SSL_SOME_SUITES_USE_MAC`, `MBEDTLS_SSL_SOME_SUITES_USE_CBC`, etc. so we should extend that pattern to `MBEDTLS_SSL_SOME_SUITES_USE_AEAD`.

Reply:
@gabor-mezei-arm